src/gatsby/models/authors/onCreateMarkdownContentNode.js

```diff
@@ -1,16 +1,50 @@
+function parseLink(input) {
+  if (typeof input === 'string') {
+    const url = new URL(input)
+    // Slice the 'www.' off of given hostnames to normalize them.
+    const hostname = url.hostname.startsWith('www.')
```

Comment:
nitpick: I would probably do a one liner with regex

Reply:
you can even remove trailing slash from the path if it exists

Reply:
I more or less completely remade the implementation that is mostly based off regex- not just become of this nitpick, but to handle edge cases I found when writing the tests.